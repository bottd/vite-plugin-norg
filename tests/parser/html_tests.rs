use insta::assert_yaml_snapshot;
use rstest::rstest;
use std::fs;
use vite_plugin_norg_parser::{extract_metadata, extract_toc, transform};

#[rstest]
#[case::basic("tests/fixtures/basic.norg")]
#[case::code_blocks("tests/fixtures/code-blocks.norg")]
#[case::headings("tests/fixtures/headings.norg")]
#[case::images("tests/fixtures/images.norg")]
#[case::links("tests/fixtures/links.norg")]
fn test_norg_fixture_files(#[case] fixture_path: &str) {
    let content = fs::read_to_string(fixture_path)
        .unwrap_or_else(|_| panic!("Failed to read {fixture_path}"));
    let ast = rust_norg::parse_tree(&content)
        .unwrap_or_else(|_| panic!("Failed to parse {fixture_path}"));

    let html = transform(&ast);
    let toc = extract_toc(&ast);

    let metadata = extract_metadata(&ast);
    assert_yaml_snapshot!(fixture_path, (html, toc, metadata));
}
