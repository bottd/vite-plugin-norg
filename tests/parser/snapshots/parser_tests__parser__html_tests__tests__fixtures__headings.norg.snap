---
source: tests/parser/html_tests.rs
expression: "(html, toc, metadata)"
---
- "<h1 id=\"h1\">H1</h1>\n<p>Content 1</p><h2 id=\"h2\">H2</h2>\n<p>Content 2</p><h3 id=\"h3\">H3</h3>\n<p>Content 3</p><h4 id=\"h4\">H4</h4>\n<p>Content 4</p><h5 id=\"h5\">H5</h5>\n<p>Content 5</p><h6 id=\"h6\">H6</h6>\n<p>Content 6</p>"
- - level: 1
    title: H1
    id: h1
  - level: 2
    title: H2
    id: h2
  - level: 3
    title: H3
    id: h3
  - level: 4
    title: H4
    id: h4
  - level: 5
    title: H5
    id: h5
  - level: 6
    title: H6
    id: h6
- title: "\"Headings\""
